### DISCUSSION TITLE
ESLint couldn't determine the plugin "react" uniquely.

### DISCUSSION THREAD

**Discussion by faizan2077** [eslint/eslint#18909] [Q&A]
I have upgraded my nodejs version for my react app from 16.20.2 to the 17.0.1 
I'm trying to run lint in my code and getting this error:

```
Oops! Something went wrong! :(

ESLint: 7.32.0

ESLint couldn't determine the plugin "react" uniquely.

- /Users/apple/Desktop/amplify/amplify.webapp/node_modules/eslint-plugin-react/index.js (loaded in ".eslintrc » eslint-config-airbnb » /Users/apple/Desktop/amplify/amplify.webapp/node_modules/eslint-config-airbnb/rules/react-a11y.js")
- /Users/apple/Desktop/amplify/amplify.webapp/node_modules/eslint-config-react-app/node_modules/eslint-plugin-react/index.js (loaded in ".eslintrc » eslint-config-react-app » /Users/apple/Desktop/amplify/amplify.webapp/node_modules/eslint-config-react-app/base.js")

Please remove the "plugins" setting from either config or remove either plugin installation.
```

**Comment by amareshsm** [comment]
It looks like ESLint has detected two different instances of the eslint-plugin-react plugin.

How the Plugin Instances are Loaded:

1. The first instance (eslint-plugin-react) is being used directly or through a configuration like eslint-config-airbnb.
The second instance (eslint-plugin-react) is being loaded indirectly through eslint-config-react-app, which also has its own dependencies.

2. When ESLint encounters multiple instances of the same plugin, it can cause conflicts and confusion about which version or configuration should be applied.



Ensure that your ESLint configuration is consistent and does not redundantly include multiple plugins. For example, if you’re using eslint-config-airbnb, you might not need to separately include eslint-config-react-app.

Review and update your .eslintrc file(s) to ensure they are correctly configured. If you're having trouble resolving the issue, please share your ESLint configuration file.

**Reply by faizan2077** [reply]
Here is my eslint config file :
```
{
  "env": {
    "browser": true,
    "jest": true,
    "es6": true
  },
  "extends": ["airbnb", "prettier", "plugin:react/recommended"],
  "plugins": ["prettier", "react", "react-hooks"],
  "parserOptions": {
    "ecmaVersion": 2020,
    "sourceType": "module"
  },
  "ignorePatterns": ["**/ErrorBoundary/*.js"],
  "settings": {
    "import/resolver": {
      "node": {
        "paths": ["src"]
      }
    }
  },
  "rules": {
    "jsx-a11y/label-has-associated-control": 0,
    "jsx-a11y/label-has-for": 0,
    "prettier/prettier": 0,
    "no-eval": "error",
    "no-bitwise": "error",
    "no-console": "warn",
    "no-unused-vars": 0,
    "react/forbid-prop-types": 0,
    "react/no-multi-comp": "warn",
    "react/react-in-jsx-scope": 0,
    "react/require-default-props": 0,
    "react/jsx-filename-extension": 0,
    "react/jsx-props-no-spreading": 0,
    "react/destructuring-assignment": 0,
    "react/jsx-no-bind": 0,
    "import/no-anonymous-default-export": 0,
    "react-hooks/exhaustive-deps": 0,
    "react/no-array-index-key": 0,
    "react/no-unused-prop-types": 0,
    "import/first": 0,
    "no-use-before-define": 0,
    "no-restricted-syntax": 0,
    "no-unused-expressions": 0,
    "react/jsx-key": "error"
  }
}

```

**Reply by amareshsm** [reply]
Since eslint-config-airbnb already includes the eslint-plugin-react, you don't need to explicitly declare it in the plugins section. You can safely remove "react" from the plugins array.

Update the ESLint configuration, then remove the node_modules directory. Reinstall the dependencies and check if the issue is resolved.

**Reply by faizan2077** [reply]
But from the above code config my linter in working fine, but there is another issue raised, I'm unable to build my react app, here is what I got:
```
[eslint] Plugin "react" was conflicted between ".eslintrc" and ".eslintrc » eslint-config-airbnb » /Users/apple/Desktop/amplify/amplify.webapp/node_modules/eslint-config-airbnb/rules/react-a11y.js".
```

**Reply by amareshsm** [reply]
Are you using the Create React App?
